ui/pages/Applications.tsx

```diff
@@ -1,50 +1,132 @@
+import { Checkbox } from "@material-ui/core";
 import * as React from "react";
 import styled from "styled-components";
 import Button from "../components/Button";
 import DataTable from "../components/DataTable";
 import Flex from "../components/Flex";
+import Icon, { IconType } from "../components/Icon";
 import Link from "../components/Link";
 import Page, { TitleBar } from "../components/Page";
 import useApplications from "../hooks/applications";
 import { Application } from "../lib/api/applications/applications.pb";
 import { PageRoute } from "../lib/types";
 import { formatURL } from "../lib/utils";
-
 type Props = {
   className?: string;
 };
 
 function Applications({ className }: Props) {
   const [applications, setApplications] = React.useState<Application[]>([]);
+  const [sort, setSort] = React.useState("Name");
+  const [reverseSort, setReverseSort] = React.useState(false);
   const { listApplications, loading } = useApplications();
 
   React.useEffect(() => {
     listApplications().then((res) => setApplications(res as Application[]));
   }, []);
 
+  type labelProps = { label: string };
+  function SortableLabel({ label }: labelProps) {
```

Comment:
I would expect we would use this on every `<DataTable />`. What about putting this functionality there?